ontotext-yasgui-web-component/src/components/tooltip/yasgui-tooltip.tsx

```diff
@@ -0,0 +1,87 @@
+import {Component, Element, h, Host, Prop, Watch} from '@stencil/core';
+import tippy, {Instance, Placement} from 'tippy.js';
+
+@Component({
+  tag: 'yasgui-tooltip',
+  styleUrl: 'yasgui-tooltip.scss',
+  shadow: false,
+})
+export class YasguiTooltip {
+
+  @Element() el: HTMLElement;
+
+  @Prop() dataTooltip: string;
+
+  @Prop() placement: string;
+
+  @Prop() showOnClick: false;
+
+  @Watch('dataTooltip')
+  configurationChanged() {
+    this.update();
+  }
+
+  private tooltip: Instance;
+
+  private showFunction;
+  private hideFunction;
+
+  componentDidLoad() {
+    // As documentation said "The @Watch() decorator does not fire when a component initially loads."
+    // yasgui instance will not be created if we set configuration when component is loaded, which
+    // will be most case of the component usage. So we call the method manually when component is
+    // loaded. More info https://github.com/TriplyDB/Yasgui/issues/143
+    this.init();
+  }
+
+  disconnectedCallback() {
+    this.destroy();
+  }
+
+  render() {
+    return (
+      <Host class="yasgui-tooltip">
+        <slot />
+      </Host>
+    );
+  }
+
+  private update() {
+    this.tooltip.setContent(this.dataTooltip);
+  }
+
+  private init() {
```

Comment:
That's what I'm talking about. Adding these buttons inside we start introducing additional dependencies which we should watch out and additional code that we have to support. :(